src/common/varint.h

```diff
@@ -78,6 +78,22 @@ namespace tools {
     dest++;			/* Seems kinda pointless... */
   }
 
+  /*! \brief writes a varint to a fixed-size uchar array of size N.
+   */
+  template<typename T, std::size_t N>
+  void encode_varint(T t, unsigned char (&out)[N]) {
+    static_assert(std::is_integral<T>::value, "");
+    static_assert(std::is_unsigned<T>::value, "");
+    static_assert((sizeof(T) * 8 + 6) / 7 <= N, "");  //output array must be large enough to store any varint encoding of 't'
+    for (std::size_t i = 0; i < N && t; ++i) {
```

Comment:
This will write nothing if the value passed is 0. I don't think this is the intent. A unit test checking it writes the same as the existing varint writer (or that the reader gets back the same values) would find this.